cmd/kube-proxy/app/server.go

```diff
@@ -1086,18 +1095,12 @@ func detectNodeIPs(ctx context.Context, rawNodeIPs []net.IP, bindAddress string)
 
 // getNodeIP returns IPs for the node with the provided name.  If
 // required, it will wait for the node to be created.
-func getNodeIPs(ctx context.Context, client clientset.Interface, name string) []net.IP {
+func getNodeIPs(ctx context.Context, nodeLister corelisters.NodeLister, name string) []net.IP {
 	logger := klog.FromContext(ctx)
 	var nodeIPs []net.IP
-	backoff := wait.Backoff{
-		Steps:    6,
-		Duration: 1 * time.Second,
-		Factor:   2.0,
-		Jitter:   0.2,
-	}
 
-	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
-		node, err := client.CoreV1().Nodes().Get(ctx, name, metav1.GetOptions{})
+	err := wait.PollUntilContextCancel(ctx, 1*time.Second, true, func(context.Context) (bool, error) {
+		node, err := nodeLister.Get(name)
```

Comment:
This change means that we're switching from consistent get, to potentially stale read.
[We will fix that hopefully in 1-2 releases with ConsistentReadFromCache and/or WatchList graduation to GA, but that's not there yet].

I'm not well positioned if that will cause issues or not, but if not, I would like to see some comment explaining why.

Reply:
@aojea mentioned 

> informer goes to watch cache, there is an event handler that runs later that detects if the node is stale and restart the apiserver as the Get() can also present problems wtih spot vms per example

I might miss it but only see the check CIDR for staleness

Reply:
I need to revisit this logic, the node IPs seems to be used only for the primary ip family and for the loadbalancer source ranges , that now we know some times may be not working as expected as it was falling back to 127.0.0.1

/hold

Reply:
Can we just add the check with node IPs, which will make proxy restart if unmatched and we can proceed this PR?

Reply:
/hold cancel

added as independent check so we can remove it once the features wojtek mentions are fixed

Reply:
It doesn't matter if we get stale data here; kube-proxy will exit if the node object changes later.
(And we don't expect the node IPs to change often anyway.)